docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,171 @@
+# Smart Switch HA – SONiC MGMT Test Plan
+
+## Revision History
+
+| Date       | Author        | Description                      |
+|------------|---------------|----------------------------------|
+| 2024-02-06 | Jing Zhang   | Initial draft                    |
+
+## Table of Contents
+
+- [Smart Switch HA – Test Plan](#smart-switch-ha--test-plan)
+  - [Revision History](#revision-history)
+  - [Scope](#scope)
+  - [Topology](#topology)
+  - [Test Methodology](#test-methodology)
+  - [Test Plan](#test-plan)
+    - [Downstream (T2->T0) Traffic Verification](#downstream-t2t0-traffic-verification)
+      - [Module 1 	Normal OP and Planned Events](#module-1--normal-op-and-planned-events)
+      - [Module 2 	BFD state pinned](#module-2--bfd-state-pinned)
+      - [Module 3 	HA state pinned by upstream service](#module-3--ha-state-pinned-by-upstream-service)
+      - [Module 4 	Link Failures](#module-4--link-failures)
+      - [Module 5 	Critical Process Crash](#module-5--critical-process-crash)
+      - [Module 6 	Power down and hardware failure](#module-6--power-down-and-hardware-failure)
+      - [Module 7 	Operations](#module-7--operations)
+    - [Upstream (T0->T2) Traffic Verification](#upstream-t0t2-traffic-verification)
+      - [Module 1 Normal OP](#module-1-normal-op)
+  - [Test Utilities](#test-utilities)
+
+## Scope 
+This document proposes solutions for Smart Switch High-Availability test plans. The document will cover smart switch test scenarios.
+
+The goal of this test plan is to verify HA state machine behavior in normal operation scenarios and network failure scenarios with [t1-smartswitch-ha](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/vars/topo_t1-smartswitch-ha.yml) topology. Both control plane and data plane will need to be verified in the test cases. 
+
+## Topology
+
+Tests will run on a sonic-mgmt testbed. The key components will be:
+1. Test server
```

Comment:
nit: installing a markdown-link extension in vscode can help identify and fix many markdown format issues.

Reply:
Thanks for the tip. Addressed.

Reply:
hi @zjswhhh , just in case - I am not seeing update on this PR, so I wonder if you forgot to push the new commits.